src/nvim/api/buffer.c

```diff
@@ -738,6 +742,223 @@ ArrayOf(Integer, 2) nvim_buf_get_mark(Buffer buffer, String name, Error *err)
   return rv;
 }
 
+/// Returns extended mark info for a given mark identifier
+/// If no mark is found, returns an empty list.
+///
+/// If an invalid namespace or buffer is given, an error will be
+/// raised and 0 returned.
```

Comment:
Is "0 returned" still correct? Above it says "If no mark is found, returns an empty list".

Reply:
also, if error is raised, sentinel value must be returned (empty list) and not mentioned in docs, as it is redundant. Also "buffer is given an error will raised" should not be mentioned, this behavior is shared among all buffer functions.

Reply:
Yes the documentation is correct, if an error occurs, 0 is currently returned

Reply:
No, `ARRAY_DICT_INIT` is returned. And it is only returned as a detail of C API, semantically nothing is returned, and so it should not be mentioned at each and every function...

Reply:
Should i mention error on invalid namespace?

Reply:
perhaps it is enough to mention at `nvim_buf_set_extmark`, as that is the first action any plugin meaningfully would do.